main.js

```diff
@@ -276,25 +283,130 @@ function displayLargeCard() {
+  $cartListContainer.appendChild($cartItemsContainer)
+
+  var $cartItemDiv = document.createElement('div')
+  $cartItemDiv.classList.add('container')
+  $cartItemsContainer.appendChild($cartItemDiv)
+
+  var $cartImageContainer = document.createElement('div')
+  $cartImageContainer.classList.add('.large-card-image-container')
+  $cartItemDiv.appendChild($cartImageContainer)
+
+  var $cartImage = document.createElement('img')
+  $cartImageContainer.appendChild($cartImage)
+  $cartImage.setAttribute('src', item.imageUrl)
+  $cartImage.classList.add('cart-image')
+
+  var $cartTextImageContainer = document.createElement('div')
+  $cartItemDiv.appendChild($cartTextImageContainer)
+  $cartTextImageContainer.classList.add('cart-text-image-container')
+
+  var $cartItemName = document.createElement('h5')
+  $cartTextImageContainer.appendChild($cartItemName)
+  $cartItemName.textContent = item.name
+
+  var $cartItemBrand = document.createElement('h5')
+  $cartTextImageContainer.appendChild($cartItemBrand)
+  $cartItemBrand.textContent = item.brand
+
+  var $cartItemPrice = document.createElement('h5')
+  $cartTextImageContainer.appendChild($cartItemPrice)
+  $cartItemPrice.textContent = item.price
+
+  return $cartListContainer
+}
+
+var cartObjects = app.cart.items
+
+function renderCartPage(cartObjects) {
+  var $cartHeader = document.createElement('h1')
+  var $cartListContainer = document.createElement('div')
+  $cartListContainer.classList.add('cartListContainer')
+  $cartHeader.textContent = 'Cart'
+  $cartHeader.classList.add('headerText')
+  $cartListContainer.appendChild($cartHeader)
+
+  for (var k = 0; k < cartObjects.length; k++) {
+    $cartListContainer.appendChild(createCartItem(app.cart.items[k]))
+  }
+
+  var sum = 0
+  for (var i = 0; i < cartObjects.length; i++) {
+    sum += cartObjects[i].price
+  }
+
+  var $cartTotalContainer = document.createElement('div')
+  $cartTotalContainer.classList.add('container', 'cart-total-container')
+  $cartListContainer.appendChild($cartTotalContainer)
+
+  var $priceTotal = document.createElement('h4')
+  $priceTotal.textContent = 'Total: $' + sum
+  $cartTotalContainer.appendChild($priceTotal)
+
+  var $itemTotal = document.createElement('h4')
+  $itemTotal.textContent = cartObjects.length + ' Item(s)'
+  $cartTotalContainer.appendChild($itemTotal)
+
+  var $cartContinueShoppingContainer = document.createElement('div')
+  $cartContinueShoppingContainer.classList.add('container', 'cart-total-container')
+  $cartListContainer.appendChild($cartContinueShoppingContainer)
+
+  var $continueShoppingButton = document.createElement('a')
+  $cartContinueShoppingContainer.appendChild($continueShoppingButton)
+  $continueShoppingButton.classList.add('btn', 'btn-primary')
+  $continueShoppingButton.textContent = 'Continue Shopping'
+
+  $continueShoppingButton.addEventListener('click', returnToCatalog)
+
+  return $cartListContainer
+}
+
+function returnToCatalog() {
+  $catalog.innerHTML = ''
```

Comment:
You shouldn't need to empty the views in this function. Your `renderAppState` function should be smart enough to empty out the different views when appropriate.

Reply:
![screen shot 2018-07-05 at 1 10 58 pm](https://user-images.githubusercontent.com/13921070/42345657-e86ae378-8054-11e8-81d6-19d902cbdfc2.png)
![screen shot 2018-07-05 at 1 11 39 pm](https://user-images.githubusercontent.com/13921070/42345689-0516c35c-8055-11e8-9bce-1dd70cc090ce.png)